igvc_platform/src/imu/YostLabDriver.cpp

```diff
@@ -170,56 +171,36 @@ const std::string YostLabDriver::getMIMode()
   return ret_buf;
 }
 
-//! Run the serial sync
 void YostLabDriver::run()
 {
-  this->flush();
-  this->setTimeout(500);
-  this->SerialWriteString(SET_AXIS_DIRECTIONS);
-  this->SerialWriteString(SET_STREAMING_SLOTS);
-  this->SerialWriteString(COMMIT_SETTINGS);
-
-  ros::Duration(0.02).sleep();
-
-  // print/debug statements
-  software_version_ = this->getSoftwareVersion();
-  axis_direction_ = this->getAxisDirection();
-  std::string euler = this->getEulerDecomp();
-  calibration_mode_ = this->getCalibMode();
-  mi_mode_ = this->getMIMode();
+  setAndCheckIMUSettings();
 
   // Performs auto-gyroscope calibration. Sensor should remain still while samples are taken.
   this->startGyroCalibration();
 
-  // commit settings and start streaming!
+  // commit settings and start streaming
   this->flush();
-  this->SerialWriteString(SET_STREAMING_TIMING_5_MS);
-  this->SerialWriteString(START_STREAMING);
+  this->serialWriteString(SET_STREAMING_TIMING_5_MS);
+  this->serialWriteString(START_STREAMING);
 
   ros::Rate loop_rate(20);  // 20Hz
 
   int line_num_ = 0;
-  sensor_msgs::Imu imu_msg_;
-  imu_msg_.header.seq = 0;
-  std::vector<double> parsed_val_;
+  std::vector<double> parsed_val;
 
-  // orientation correction matrices in 3x3 row-major format and quaternion
-  Eigen::Matrix<double, 3, 3, Eigen::RowMajor> correction_mat(imu_orientation_correction_.data());
-  Eigen::Quaternion<double> correction_mat_quat(correction_mat);
-  tf::Quaternion rot = tf::createQuaternionFromYaw(orientation_rotation_);
   while (ros::ok())
   {
-    while (this->Available() > 0)
+    while (this->available() > 0)
```

Comment:
I still think that this could be made cleaner, ie. this entire block could be refactored away into a function or something